test/integration/order.behavior.t.sol

```diff
@@ -578,6 +578,45 @@ contract OrderBehaviorTest is Test, ConsolidatedEvents {
         account.executeConditionalOrder({_conditionalOrderId: 0});
     }
 
+    function test_ExecuteConditionalOrder_MarketIsPaused() public {
+        // place conditional order for sAUDPERP market
+        uint256 conditionalOrderId = placeConditionalOrder({
+            marketKey: sAUDPERP,
+            marginDelta: int256(AMOUNT),
+            sizeDelta: int256(AMOUNT),
+            targetPrice: 0,
+            conditionalOrderType: IAccount.ConditionalOrderTypes.LIMIT,
+            desiredFillPrice: 0,
+            reduceOnly: false
+        });
+
+        // define SystemStatus contract
+        ISystemStatus systemStatus = ISystemStatus(account.systemStatus());
+
+        // fetch owner address of SystemStatus contract
+        (bool success, bytes memory response) =
+            address(systemStatus).call(abi.encodeWithSignature("owner()"));
+        address systemStatusOwner =
+            success ? abi.decode(response, (address)) : address(0);
+
+        // add owner to access control list so they can suspend perpsv2 market
+        vm.startPrank(systemStatusOwner);
+        systemStatus.updateAccessControl({
+            section: bytes32("Futures"),
+            account: systemStatusOwner,
+            canSuspend: true,
+            canResume: true
+        });
+
+        // suspend sAUDPERP market
+        systemStatus.suspendFuturesMarket({marketKey: sAUDPERP, reason: 69});
+        vm.stopPrank();
+
+        // assert conditional order cannot be executed due to paused market
+        (bool canExecute,) = account.checker(conditionalOrderId);
+        assert(!canExecute);
```

Comment:
**Question:** Are we able to mock and test Gelatos behaviour on our end? I am just wondering how we can be sure the behaviour will be as expected.

Reply:
Well, other than mocking calls from Gelato to our contract, there is not much else we need to do. There are tests for ensuring the task is registered by Gelato when users create conditional orders. Furthermore, we checked that the task which was registered identifies the `checker` as the contract's `RESOLVER` (i.e., a function that tells Gelato whether or not to execute the task at that moment in time).

You can see [here](https://github.com/Kwenta/margin-manager/blob/362dd2c61d85ed80abd05312fe56a270d77d5246/test/integration/order.behavior.t.sol#L776) that we also mock what it would be like when Gelato can call `exec` after `checker` affirmed the task can be executed.